### PR TITLE
refector: remove pause/result from provider interface

### PR DESCRIPTION
**Pull request by richardcase**
**What this PR does / why we need it**:

As we are not supporting pausing / resuming microvms at this stage these
functions have been removed from the microvm provider interface. The
firecracker implementation has these removed as well and any tests have
been updated.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: infrastructure/firecracker/provider.go [hunk 1]
```diff
@@ -71,19 +68,15 @@ func (p *fcProvider) Start(ctx context.Context, vm *models.MicroVM) error {
...
-}
-
-// Resume will resume a paused microvm.
-func (p *fcProvider) Resume(ctx context.Context, id string) error {
-	return errNotImplemeted
-}
-
-// Stop will stop a paused or running microvm.
+// Stop will stop a running microvm.
 func (p *fcProvider) Stop(ctx context.Context, id string) error {
```
Comment (Callisto13) [inline infrastructure/firecracker/provider.go:72]: are we using stop anywhere?

Comment (Callisto13) [reply infrastructure/firecracker/provider.go:72]: can we not just remove it if it serves no purpose?

Comment (richardcase) [reply infrastructure/firecracker/provider.go:72]: Why not. I did make me wonder if we should be doing different things in stop and delete. For example, in `Stop` should we stop the firecracker process and then in `Delete` we delete the files. But lets create a ticket for follow up and remove stop for the time being. wdyt?

Comment (Callisto13) [reply infrastructure/firecracker/provider.go:72]: sounds good 👍

**Review by Callisto13 — APPROVED** [review]

File: infrastructure/firecracker/provider.go (hunk 1 above)
Comment (richardcase) [reply infrastructure/firecracker/provider.go]: Added #330

**Pull request merged**